### PR TITLE
Fix map errors with new LEL 2.2.1

### LINKED ISSUE #7381: Sidebar map layers not working
### Please describe the problem (or idea)

> What happened just before the problem occurred? Or what problem could this idea solve?

The sidebar map can't seem to find the overlay maps. 

> What did you expect to see that you didn't?

Expected to see layer data on the map. It throws the following error instead:
`
Uncaught TypeError: Cannot read property '_leaflet_id' of undefined
    at stamp (LeafletEnvironmentalLayers.js:12112)
    at NewClass.removeLayer (LeafletEnvironmentalLayers.js:18597)
    at NewClass.onRemove (LeafletEnvironmentalLayers.js:29442)
    at NewClass.removeLayer (LeafletEnvironmentalLayers.js:18602)
    at NewClass._onInputClick (LeafletEnvironmentalLayers.js:17205)
    at HTMLInputElement.handler (LeafletEnvironmentalLayers.js:14320)
`

### Please show us where to look

https://publiclab.org/notes/warren/11-16-2019/mashapaug-pond

### What's your PublicLab.org username?

christie_reni

> This can help us diagnose the issue: 

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by nstjean**
Fixes #7381 (<=== Add issue number here)

* Fixes the errors that were still happening after the new LEL version
* Using the `display` param in LEL
* Changes the peopleLeaflet inline map over to using the new `PLpeople` layer in LEL
* Cleaned up /people, it also uses the peopleLeaflet template
* Fixes a bug where some new code from fix #7386 was interfering with maps
* Now showing people (using PLpeople layer) and site content on /map
* Fixed a bug where layer marker popups weren't appearing when clicked on inline maps, but were appearing on /people and /map
* Used webmock to stub out the api used for PLpeople layer - it is currently very slow and was causing test failures

/map showing popup for a content node
FireShot Capture 302 - 🎈 Public Lab_ Map - localhost

/map showing popup for a person
FireShot Capture 301 - 🎈 Public Lab_ Map - localhost

/wiki/inline-maps
FireShot Capture 300 - 🎈 Public Lab_ Inline Maps - localhost

/people
FireShot Capture 299 - 🎈 Public Lab_ List - localhost

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by nstjean** [comment]
The test/system/post_test.rb passes on my local. It's throwing an error in travis. Have to figure out why!

FireShot Capture 303 - Job #16776 5 - publiclab_plots2 - Travis CI - travis-ci org

**Comment by nstjean** [comment]
Ok so commenting out the PLpeople layer made the tests pass. I suspect it's because that API takes so long to load. I'm going to try to stub it out for the test.

File: test/test_helper.rb [hunk 1]
```diff
@@ -37,3 +39,12 @@ def request_host
...
 end
+
+
+WebMock.allow_net_connect!
+WebMock.stub_request(:any, "publiclab.org/api/srch/nearbyPeople")
+  .to_return(
+    body: "{items:[]}",
+    status: 200,
+    :headers => {"Content-Type"=> "application/json"}
+  )
```
Comment (nstjean) [inline test/test_helper.rb]: I think this worked!

**Comment by nstjean** [comment]
Ready for review!

**Comment by cesswairimu** [comment]
@sagarpreet-chadha checkout these.. Really cool

File: app/views/map/_peopleLeaflet.html.erb [hunk 2]
```diff
@@ -7,131 +7,46 @@
...
 
     var setHash = 1;
     <% if defined? url_hash != nil and url_hash == 0 %>
       setHash = 0;
     <% end %>
 
     var options<%= unique_id %> = {
-      mainContent: "people",
-      setHash: setHash
+      layers: ["PLpeople"]
```
Comment (sagarpreet-chadha) [inline app/views/map/_peopleLeaflet.html.erb]: YESSS

File: test/test_helper.rb (hunk 1 above)
Comment (sagarpreet-chadha) [reply test/test_helper.rb]: I think you can mock any testing framework now 😄 🎉 !!!

**Comment by sagarpreet-chadha** [comment]
I am so happy to see PLpeople to go LIVE!!! This layer is one of the best layer i had the opportunity to work on. I think i can take a session on how i implemented this heatmap if you guys are interested 😄 .
Enough of self promotion :P

Thanks @nstjean  , @cesswairimu for completing this 🎉

**Comment by jywarren** [comment]
🎉 Broke out a couple follow-ups here! 

https://github.com/publiclab/leaflet-blurred-location-display/issues/98 and a little simpler still: https://github.com/publiclab/leaflet-blurred-location-display/issues/97

**Pull request merged**